multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/heancms/HeanCms.kt

```diff
@@ -122,49 +122,84 @@ abstract class HeanCms(
                 .orEmpty()
         )
 
+        val searchPayloadObj = HeanCmsSearchPayloadDto(
+            term = query
+        )
+
         val payload = json.encodeToString(payloadObj).toRequestBody(JSON_MEDIA_TYPE)
+        val searchPayload = json.encodeToString(searchPayloadObj).toRequestBody(JSON_MEDIA_TYPE)
 
         val apiHeaders = headersBuilder()
             .add("Accept", ACCEPT_JSON)
             .add("Content-Type", payload.contentType().toString())
             .build()
 
+        if (query.isNotBlank()) {
+            val apiUrl = "$apiUrl/series/search".toHttpUrl().newBuilder()
+                .addQueryParameter("q", "search")
+                .toString()
+            return POST(apiUrl, apiHeaders, searchPayload)
+        }
+
         val apiUrl = "$apiUrl/series/querysearch".toHttpUrl().newBuilder()
-            .addQueryParameter("q", query)
+            .addQueryParameter("q", "filter")
             .toString()
 
         return POST(apiUrl, apiHeaders, payload)
     }
 
     override fun searchMangaParse(response: Response): MangasPage {
         val json = response.body?.string().orEmpty()
-        val query = response.request.url.queryParameter("q").orEmpty()
+        val searchType = response.request.url.queryParameter("q").orEmpty()
+
+        if (searchType == "search") {
+            val result = json.parseAs<List<HeanCmsSearchDto>>()
+            val mangaList = result
+                .filter { it.type == "Comic" }
+                .map {
+                    getSearchMangas(it.slug.replace(TIMESTAMP_REGEX, ""), it.title)
+                }
 
-        if (json.startsWith("{")) {
-            val result = json.parseAs<HeanCmsSearchDto>()
-            var mangaList = result.data.map { it.toSManga(apiUrl, coverPath) }
+            fetchAllTitles()
 
-            if (query.isNotBlank()) {
-                mangaList = mangaList.filter { it.title.contains(query, ignoreCase = true) }
-            }
+            return MangasPage(mangaList, false)
+        }
+
+        if (json.startsWith("{")) {
+            val result = json.parseAs<HeanCmsQuerySearchDto>()
+            val mangaList = result.data.map { it.toSManga(apiUrl, coverPath) }
 
             fetchAllTitles()
 
             return MangasPage(mangaList, result.meta?.hasNextPage ?: false)
         }
 
-        var mangaList = json.parseAs<List<HeanCmsSeriesDto>>()
+        val mangaList = json.parseAs<List<HeanCmsSeriesDto>>()
             .map { it.toSManga(apiUrl, coverPath) }
 
-        if (query.isNotBlank()) {
-            mangaList = mangaList.filter { it.title.contains(query, ignoreCase = true) }
-        }
-
         fetchAllTitles()
 
         return MangasPage(mangaList, hasNextPage = false)
     }
 
+    private fun getSearchMangas(vSlug: String, vTitle: String): SManga {
```

Comment:
Make it have a more descriptive name and name the parameters better as well.

```suggestion
    private fun createSearchResult(slug: String, title: String): SManga {
```